arrow/obsidianmodule.c

```diff
@@ -472,6 +477,11 @@ _invoke_obsidian(PyObject *self, PyObject *args) {
   Py_XDECREF(substrates_indexes_array);
   Py_XDECREF(substrates_array);
 
+  // if something went wrong throw an error
+  if (obsidian == NULL) {
+    return NULL;
+  }
```

Comment:
Good catch on calling deref before this exit, however now there's no need to conditionally return NULL before returning NULL unless "throw an error" is supposed be a TODO for explicitly raising an error. Will returning NULL make the Python C API raise an error?

Reply:
Yes, returning `NULL` from a C function indicates to the python api that there is an error. All of these python interfacing functions expect some kind of non-NULL `PyObject *`.